packages/dds/map/src/directory.ts

```diff
@@ -1363,7 +1515,35 @@ class SubDirectory extends TypedEventEmitter<IDirectoryEvents> implements IDirec
 	 */
 	public subdirectories(): IterableIterator<[string, IDirectory]> {
 		this.throwIfDisposed();
-		return this._subdirectories.entries();
+		const ackedSubdirsInOrder = this.ackedCreationSeqTracker.keys();
+		const localSubdirsInOrder = this.localCreationSeqTracker.keys(
+			(key) => !this.ackedCreationSeqTracker.has(key),
+		);
+
+		const subdirNames = [...ackedSubdirsInOrder, ...localSubdirsInOrder];
+
+		assert(
+			subdirNames.length === this._subdirectories.size,
+			"The count of keys for iteration should be consistent with the size of actual data",
+		);
+
+		const entriesIterator = {
```

Comment:
Is there a TODO task to track handling changes to the map while iterating?

Reply:
I can create one and think about it